Failing "make doc_gram_rsts" deletes orderedGrammar

make/dune should not be deleting files here!  It's particularly unpleasant to lose your changes because changes to orderedGrammar are often exactly why you're making this target.

```
$ make doc_gram_rsts
     :
Warning: Nonterminal where not included in .rst files
TOTAL Nonterminals not included = 71
1 error(s)
Makefile.doc:305: recipe for target 'doc/tools/docgram/orderedGrammar' failed
make[1]: *** [doc/tools/docgram/orderedGrammar] Error 1
make[1]: *** Deleting file 'doc/tools/docgram/orderedGrammar'
```

Attn: @Zimmi48

--- a/Makefile.doc
+++ b/Makefile.doc
@@ -305,6 +305,8 @@ doc/tools/docgram/orderedGrammar doc/tools/docgram/updated_rsts: doc/tools/docgr
 	$(SHOW)'DOC_GRAM_RSTS'
 	$(HIDE)$(DOC_GRAM) $(DOCGRAMWARN) -check-cmds -check-tacs $(DOC_MLGS) $(DOC_RSTS)
 
+.PRECIOUS: doc/tools/docgram/orderedGrammar
+
 doc/tools/docgram/updated_rsts: doc/tools/docgram/orderedGrammar
 
 .PHONY: doc_gram doc_gram_verify doc_gram_rsts

